src/CloudSearchDomain/CloudSearchDomainFactory.php

```diff
@@ -26,7 +26,7 @@ protected function addDefaultArgs(&$args)
         if (strpos($args['endpoint'], 'http') !== 0) {
             $args['endpoint'] = Url::buildUrl([
                 'host'   => $args['endpoint'],
-                'scheme' => $args['scheme'],
+                'scheme' => 'https',
```

Comment:
Looks like I was wrong about this change.

This function is now invoked *before* default parameters are merged in. This function relied on the fact that "scheme" was set to "https" by default.

I was thinking about updating this function, but now I'm wondering if we should even do this. "endpoint" is supposed to be set to an absolute URL. This code here attempts to create a URL using a scheme and treat the endpoint as a host. No other client does this, and I'd like to add clarifications to the "endpoint" config option to clarify that it must be absolute. Does that sound OK?